src/colorizer/ColorizeCanvas.ts

```diff
@@ -453,17 +437,18 @@ export default class ColorizeCanvas implements IRenderCanvas {
     this.onFrameLoadCallback = callback;
   }
 
-  public setParams(params: RenderCanvasStateParams): void {
+  public async setParams(params: RenderCanvasStateParams): Promise<void> {
```

Comment:
Made async for the situation where a parent caller (`CanvasOverlay`) needs to wait for changes to be applied before re-rendering.